File: test-suite/golang/security/hardcoded_secrets_clean.go
package security

import "os"

const publicBaseURL = "https://example.com"

var authConfig = map[string]string{
	"SESSION_SECRET": mustEnv("SESSION_SECRET"),
	"WEBHOOK_SECRET": mustEnv("WEBHOOK_SECRET"),
}

type ProviderConfig struct {
	ClientSecret string
}

func providerConfig() ProviderConfig {
	return ProviderConfig{
		ClientSecret: mustEnv("OAUTH_CLIENT_SECRET"),
	}
}

func accessToken() string {
	return mustEnv("ACCESS_TOKEN")
}

func jwtSecret() string {
	if value, ok := os.LookupEnv("JWT_SECRET"); ok && value != "" {
		return value
	}
	panic("JWT_SECRET must be configured")
}

func settings() map[string]string {
	settings := map[string]string{}
	settings["API_KEY"] = mustEnv("API_KEY")
	settings["THEME_VARIANT"] = "dark"
	settings["PUBLIC_BASE_URL"] = publicBaseURL
	return settings
}

func mustEnv(key string) string {
	value, ok := os.LookupEnv(key)
	if !ok || value == "" {
		panic("missing required environment variable: " + key)
	}
	return value
}

File: test-suite/python/security/hardcoded_secrets_clean.py
import os
from os import environ, getenv


def required_env(name: str) -> str:
    value = os.environ[name]
    if not value:
        raise RuntimeError(f"Missing {name}")
    return value


SECRET_KEY = required_env("SECRET_KEY")
STRIPE_SECRET_KEY: str = required_env("STRIPE_SECRET_KEY")

auth_config = {
    "JWT_SECRET": required_env("JWT_SECRET"),
    "OAUTH_CLIENT_SECRET": required_env("OAUTH_CLIENT_SECRET"),
    "SESSION_SECRET": os.environ["SESSION_SECRET"],
    "WEBHOOK_SECRET": getenv("WEBHOOK_SECRET"),
}

settings = {}
settings["API_KEY"] = environ["API_KEY"]
settings["SIGNING_SECRET"] = required_env("SIGNING_SECRET")
settings.setdefault("COOKIE_SECRET", required_env("COOKIE_SECRET"))

flask_mapping = dict(SECRET_KEY=required_env("FLASK_SECRET_KEY"))


def jwt_secret() -> str:
    return required_env("JWT_SECRET")


class ProviderConfig:
    clientSecret = required_env("PROVIDER_CLIENT_SECRET")


apiKey = required_env("MAPS_API_KEY")
display_config = {
    "THEME_VARIANT": os.getenv("THEME_VARIANT", "dark"),
    "PUBLIC_BASE_URL": os.getenv("PUBLIC_BASE_URL", "https://example.com"),
}
